contracts/Vault.vy

```diff
@@ -932,7 +935,7 @@ def _sharesForAmount(amount: uint256) -> uint256:
         return  (
             amount
             * self.totalSupply
-            / self._totalAssets()
+            / self.freeFunds()
```

Comment:
Why this change?

Reply:
Because the shares for the amount in the event of a loss should be calculated based on the free funds and not the totalAssets https://github.com/yearn/yearn-vaults/issues/390

Reply:
Okay. Should probably also adjust the `if` statement above (and cache that value too)

Reply:
👌